src/PromptMatcher.php

```diff
@@ -35,20 +35,15 @@ class PromptMatcher
      */
     public function isMatch($prompt, $subject, $lineEnding)
     {
-        // cheap line ending check before expensive regex
-        if (substr($subject, -1 * strlen($lineEnding)) != $lineEnding) {
-            return false;
-        }
 
         $matches = [];
         $callback = function ($matchesCallback) use (&$matches) {
             $matches = $matchesCallback;
             // replace matches with an empty string (remove prompt from $subject)
             return '';
         };
-        $pattern = sprintf('/%s%s$/', $prompt, $lineEnding);
 
-        $responseText = preg_replace_callback($pattern, $callback, $subject);
+        $responseText = preg_replace_callback('/'.$prompt.'/', $callback, $subject);
```

Comment:
Including the `$` at the end of the regex should make the preg less resource intensive. It shouldn't have any effect on the match as this check will have been performed on each iteration of `$subject`.